lib/widgets/post.dart

```diff
@@ -419,12 +422,63 @@ class PostWidget extends HookWidget {
               postImage()
             else if (post.post.url != null && post.post.url.isNotEmpty)
               linkPreview(),
-            if (post.post.body != null)
-              // TODO: trim content
+            if (post.post.body != null && fullPost)
               Padding(
                   padding: const EdgeInsets.all(10),
                   child:
                       MarkdownText(post.post.body, instanceHost: instanceHost)),
+            if (post.post.body != null && !fullPost)
+              LayoutBuilder(
+                builder: (context, constraints) {
+                  final span = TextSpan(
+                    text: post.post.body,
+                  );
+                  final tp = TextPainter(
+                    text: span,
+                    maxLines: 20,
+                    textDirection: Directionality.of(context),
+                  )..layout(maxWidth: constraints.maxWidth - 20);
+
+                  if (tp.didExceedMaxLines) {
+                    return Container(
+                      constraints: BoxConstraints(maxHeight: tp.height),
+                      child: Stack(
+                        alignment: Alignment.bottomCenter,
+                        children: [
+                          ClipRect(
+                            child: Align(
+                              alignment: Alignment.topCenter,
+                              heightFactor: 0.8,
```

Comment:
This seems to only cover the overflow error, it is still there and is throwing like crazy in the console. However I have also tried to do it in a different way and I have failed, it always seems to complain about the overflow. So if you are also out of ideas let's leave it like that

Reply:
Yes I also tried to remove the error by adding this code:

```
if (tp.didExceedMaxLines) {                   
  final tpFullHeight = TextPainter(
    text: span,
    textDirection: Directionality.of(context),
  )..layout(maxWidth: constraints.maxWidth - 20);
  
  final str = post.post.body.substring(
    0,
    post.post.body.length ~/
       ((tpFullHeight.height + 2 * tpFullHeight.preferredLineHeight) /
           tp.height));
```
and passing `str` as argument instead of `post.post.body` to `MarkdownText` but it is not perfect solution as not all lines of text will be of same length therefore many times it leaves out too many characters and lot of empty space is shown.

Therefore I also think let's leave it like that.